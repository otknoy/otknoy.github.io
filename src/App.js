import React from 'react'

import Header from './Header'
import Profile from './Profile'
import Portfolio from './Portfolio'
import Interest from './Interest'
import Footer from './Footer'

const state = {
  profile: {
    name: 'Naoya Otsuka',
    description: 'web engineer',
    link: {
      github: 'https://github.com/otknoy',
      twitter: 'https://twitter.com/otknoy',
      blog: 'https://otknoy.hatenablog.com'
    }
  },
  interest: [
    'Java, Kotlin, SpringBoot',
    'Python, Node.js, Go',
    'Docker, Kubernetes',
    'Microservices, Test-driven Development',
    'Information Retrieval', 
    'Human Computer Interaction'
  ],
  portfolio: [
    {
      name: 'under construction',
      link: '/'
    }
  ]
}

const Page = () => (
    <>
      <Header />

      <Profile {...state.profile}/>
      <Interest interest={state.interest}/>
      <Portfolio portfolio={state.portfolio}/>
    
      <Footer />
    </>
)

export default Page
