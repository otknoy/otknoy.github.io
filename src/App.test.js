import React from 'react'
import renderer from 'react-test-renderer'

import App from './App'

describe('render <App />', () => {
  const OriginalDate = Date
  jest.spyOn(global, 'Date').mockImplementation((arg) => {
    return arg ? new OriginalDate(arg) : new Date('2025-01-01')
  })

  test('snapshot', () => {
    const tree = renderer.create(<App />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
