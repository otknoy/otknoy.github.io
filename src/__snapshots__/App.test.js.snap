// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <App /> snapshot 1`] = `
Array [
  <header
    className="sc-eCApGN dyORaM"
  >
    <h1
      className="sc-jSFkmK jPCUDi"
    >
      <span
        className="sc-hKFyIo eCqoGD"
      >
        Naoya Otsuka
      </span>
    </h1>
  </header>,
  <section>
    <h1>
      Profile
    </h1>
    <div>
      <p>
        Naoya Otsuka
      </p>
      <p>
        web engineer
      </p>
    </div>
    <ul>
      <li>
        <a
          href="https://github.com/otknoy"
        >
          github
        </a>
      </li>
      <li>
        <a
          href="https://twitter.com/otknoy"
        >
          twitter
        </a>
      </li>
      <li>
        <a
          href="https://otknoy.hatenablog.com"
        >
          blog
        </a>
      </li>
    </ul>
  </section>,
  <section>
    <h1>
      Interest
    </h1>
    <ul>
      <li>
        Java, Kotlin, SpringBoot
      </li>
      <li>
        Python, Node.js, Go
      </li>
      <li>
        Docker, Kubernetes
      </li>
      <li>
        Microservices, Test-driven Development
      </li>
      <li>
        Information Retrieval
      </li>
      <li>
        Human Computer Interaction
      </li>
    </ul>
  </section>,
  <section>
    <h1>
      Portfolio
    </h1>
    <ul>
      <li>
        <a
          href="/"
        >
          under construction
        </a>
      </li>
    </ul>
  </section>,
  <footer
    className="sc-gKAblj iDszLg"
  >
    <h1
      className="sc-iCoHVE AscQy"
    >
      <span
        className="sc-gtssRu fBCpkM"
      >
        (C) 1988-2020 Naoya Otsuka
      </span>
    </h1>
  </footer>,
]
`;
